From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ram Muthiah <rammuthiah@google.com>
Date: Tue, 21 May 2019 18:15:10 -0700
Subject: ANDROID-Four-part-revert-of-asm-goto-usage-1-4

Revert "x86/uaccess: Dont leak the AC flag into __put_user() argument evaluation"
This reverts commit 6ae865615fc43d014da2fd1f1bba7e81ee622d1b.

Bug: 120440614
Bug: 132629930
Change-Id: I38f78e20d255ab4b33546d2e9d98f64d7e590e1d
Signed-off-by: Ram Muthiah <rammuthiah@google.com>
---
 arch/x86/include/asm/uaccess.h | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/arch/x86/include/asm/uaccess.h b/arch/x86/include/asm/uaccess.h
index 61d93f062a36..176c73385605 100644
--- a/arch/x86/include/asm/uaccess.h
+++ b/arch/x86/include/asm/uaccess.h
@@ -429,11 +429,10 @@ do {									\
 ({								\
 	__label__ __pu_label;					\
 	int __pu_err = -EFAULT;					\
-	__typeof__(*(ptr)) __pu_val = (x);			\
-	__typeof__(ptr) __pu_ptr = (ptr);			\
-	__typeof__(size) __pu_size = (size);			\
+	__typeof__(*(ptr)) __pu_val;				\
+	__pu_val = x;						\
 	__uaccess_begin();					\
-	__put_user_size(__pu_val, __pu_ptr, __pu_size, __pu_label);	\
+	__put_user_size(__pu_val, (ptr), (size), __pu_label);	\
 	__pu_err = 0;						\
 __pu_label:							\
 	__uaccess_end();					\
